pkg/mobile/types.go

```diff
@@ -27,3 +27,15 @@ type DeviceMetric struct {
 	Platform        string `json:"platform"`
 	PlatformVersion string `json:"platformVersion"`
 }
+
+func (m *Metric) Validate() (valid bool, reason string) {
+	if m.ClientId == "" {
```

Comment:
will this be an empty string if not submitted or `nil`?

Reply:
Looks like it can only be nil if it's a pointer: https://stackoverflow.com/questions/12703243/what-is-the-zero-for-string

Learning this now myself too :D

Reply:
Thanks, good to know! I have not much of an Idea about Golang ;)

Reply:
As part of decoding JASON  into a struct, if a field is omitted it is instantiated with it's 'zero' value. The zero value for a string is "".

You see later on that we check if m.Data is nil. That's because the Data field expects a type *MetricData which is a pointer as opposed to the value itself. The zero value of a pointer is nil, hence the nil check for metric.Data but empty string check for ClientId. Does that make sense @pb82 ?

Reply:
@darahayes yes makes sense, thanks for explaining that